src/refda/adm/ietf_dtnma_agent.c

```diff
@@ -2274,6 +2587,77 @@ static void refda_adm_ietf_dtnma_agent_ctrl_ensure_rule_enabled(refda_ctrl_exec_
      * |START CUSTOM FUNCTION refda_adm_ietf_dtnma_agent_ctrl_ensure_rule_enabled BODY
      * +-------------------------------------------------------------------------+
      */
+    const cace_ari_t *target      = refda_ctrl_exec_ctx_get_aparam_index(ctx, 0);
+    const cace_ari_t *ari_enabled = refda_ctrl_exec_ctx_get_aparam_index(ctx, 1);
+
+    cace_ari_t ari_result;
+    cace_ari_init(&ari_result);
+    cace_ari_set_uint(&ari_result, 0);
+
+    cace_ari_bool enabled;
+    if (cace_ari_get_bool(ari_enabled, &enabled))
```

Comment:
Consider moving this precondition above the allocation of `ari_result` to avoid leak, and adding a check on the content of target (even if the check is just `cace_ari_not_undefined()`). PR #164 adds that function and needs a review. PR #169 also adds a simple check for all parameters not-undefined.